lib/pmc_miller/puppetdb/queue_depth.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module PmcMiller
+  ##
+  # Namespace analyzers specific to PuppetDB
+  module PuppetDB
+    ##
+    # Analyze queue_depth data
+    class QueueDepth
+      attr_reader :data
+
+      ##
+      # Creates a new QueueDepth analyzer
+      #
+      # @param data [Array] queue_depth data to analyze
+      #
+      # @return [PmcMiller::PuppetDB::QueueDepth]
+      #
+      def initialize(data)
+        @data = data
+      end
+
+      ##
+      # Calculate rate of change for data.
+      #
+      # The average rate of change is calculated between the first and second
+      # halves of the data set.
+      #
+      # @return [Float]
+      #
+      def rate_of_change
+        return @r_o_c if defined?(@r_o_c)
+
+        @r_o_c = (first_half_slope + last_half_slope) / 2.0
```

Comment:
This class attribute should be documented with the class up above.  And I just am not a fan of names like r_o_c.  I think we can afford the extra characters... :)

Reply:
I have renamed the instance variable so that it is less cryptic.  However, it is not an attribute and is not accessible as such.  It is intended as an internal instance variable to cache the rate_of_change calculation.  For that reason, I would argue that it should not be publicly documented as an attribute of the class.